### PR TITLE
Zeeshan fe nav context 21

### PR DESCRIPTION
**Pull request by sk-izsk**
navbar added with context configuration
Screenshot 2020-10-16 194338

### CODE REVIEW

**Review by SsaaaadD2 — COMMENTED** [review]
Changes look good to me

File: client/src/App.jsx [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
+  const [appContext, setAppContext] = useState({});
+  const stateContext = useContext(GlobalContext);
+
+  useEffect(() => {
+    setAppContext(stateContext);
+
+    return () => {
+      setAppContext({});
+    };
+  }, [stateContext]);
```
Comment (aeciorc) [inline client/src/App.jsx]: Is this necessary? What's the advantage over using the context directly?

File: client/src/App.jsx [hunk 2]
```diff
@@ -1,25 +1,34 @@
...
 
-function App() {
   return (
-    <MuiThemeProvider theme={theme}>
-      <BrowserRouter>
+    <>
+      {!!appContext.user && appContext.user.success && <Header />}
```
Comment (aeciorc) [inline client/src/App.jsx]: I don't think user will have a param "success". You can leave that out

Comment (sk-izsk) [reply client/src/App.jsx]: my miss, it was there. will change then

File: client/src/App.jsx (hunk 1 above)
Comment (sk-izsk) [reply client/src/App.jsx]: hmm well nothing... just feel use useEffect 😋 ... at first i use directly.... if you want i can change then 😁

Comment (aeciorc) [reply client/src/App.jsx]: I think it's better to change, the results will be the same since it re-renders when the context change, right?

**Comment by aeciorc** [comment]
Ok, this can be merged after implementing the changes I mentioned

File: client/src/App.jsx (hunk 1 above)
Comment (sk-izsk) [reply client/src/App.jsx]: its basically doing same, since haven't use useEffect so far thats why just put it there. but sure, will change it then

**Pull request merged**